### PR TITLE
Final Submission

### PR DESCRIPTION
**Pull request by Cabraham1**

### CODE REVIEW

File: src/modules/displayList.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          <div class="card-content">
+          <div class="first-part">
+            <h2>${data.strMeal}</h2>
+            <div class='like-part likes-${data.idMeal}'>
+            <i class="fa-solid fa-heart like-icon" id='${data.idMeal}'></i> 
+            <span class='like-count ${data.idMeal}>' id='${data.idMeal}'>0 likes</span>
+            </div>
+          </div>
+            <div class="second-part">
+            <button class="sea-meal">Comment</button>
```
Comment (VICTOR-CHUKWU) [inline src/modules/displayList.js:20]: - [ ] in as much as you worked on the project alone, you still need to include a pop-up for comment where a user can view and make comments as it is in the design
Screenshot from 2023-03-16 23-27-28

File: src/__test__/sum.test.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import sum from '../sum.js';
+
+test('adds 1 + 2 to equal 3', () => {
+  expect(sum(1, 2)).toBe(3);
+});
```
Comment (VICTOR-CHUKWU) [inline src/__test__/sum.test.js:5]: - [ ] I really don't know what this test is for, kindly remove it

File: src/modules/itemCounter.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+const itemCounter = document.querySelector('.item-counter');
+
+const menuCounter = () => {
+  fetch(API_URL)
+    .then((response) => response.json())
+    .then((data) => {
+      const meal = data.meals;
+      const count = meal.length;
+      itemCounter.innerHTML = `(${count})`;
+    });
```
Comment (VICTOR-CHUKWU) [inline src/modules/itemCounter.js:11]: - [ ] according to project requirements kindly use async and await with the fetch method to send and receive data.

File: src/modules/likesCounter.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+const displayLikes = () => {
+  fetch(URL_API)
+    .then((response) => response.json())
+    .then((data) => {
+      data.forEach((item) => {
+        const likeCount = document.querySelector(`.likes-${item.item_id}`);
+        likeCount.querySelector('span').textContent = `${item.likes} likes`;
+      });
+    });
+};
```
Comment (VICTOR-CHUKWU) [inline src/modules/likesCounter.js:39]: - [ ] according to project requirements kindly use async and await with the fetch method to send and receive data.

File: src/modules/displayList.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  mealData.appendChild(meal);
+};
+
+const display = async () => {
+  const response = await fetch(API_URL);
+  const data = await response.json();
+  data.meals.forEach((food) => {
+    displayMeal(food);
+  });
+};
```
Comment (VICTOR-CHUKWU) [inline src/modules/displayList.js:34]: - [ ] Good use of async await method :clap:

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Cabraham1 , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
_Check the comments under the review._

- [ ] your MIT link is broken in your README file, kindly fix this
- [ ] your Livelink is not redirecting to your live page kindly check this and fix

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/__test__/sum.test.js (hunk 2 above)
Comment (Cabraham1) [reply src/__test__/sum.test.js:5]: okay sir

File: src/modules/itemCounter.js (hunk 3 above)
Comment (Cabraham1) [reply src/modules/itemCounter.js:11]: done thank you

File: src/modules/likesCounter.js (hunk 4 above)
Comment (Cabraham1) [reply src/modules/likesCounter.js:39]: functionality added

File: src/modules/displayList.js (hunk 5 above)
Comment (Cabraham1) [reply src/modules/displayList.js:34]: thanks

File: src/modules/displayList.js (hunk 1 above)
Comment (Cabraham1) [reply src/modules/displayList.js:20]: done

File: README.md [hunk 6]
```diff
@@ -40,249 +12,127 @@ After you're finished please remove all the comments and instructions!
...
 
-- [Live Demo Link](https://yourdeployedapplicationlink.com)
 
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-<!-- GETTING STARTED -->
+## Live Demo
 
-## 💻 Getting Started <a name="getting-started"></a>
+[Live demo here](https://Cabraham1.github.io/capstone-project/)
```
Comment (DuaneDave) [inline README.md:61]: As previously mentioned, the live link is still not leading to your project, instead I get a 404 page error. Kindly address this to improve the documentation.

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @Cabraham1 👋, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 6 above)
Comment (Cabraham1) [reply README.md:61]: done

File: src/modules/likesCounter.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+const displayLikes = async () => {
+  await fetch(URL_API)
+    .then((response) => response.json())
+    .then((data) => {
+      data.forEach((item) => {
+        const likeCount = document.querySelector(`.likes-${item.item_id}`);
+        likeCount.querySelector('span').textContent = `${item.likes} likes`;
+      });
```
[the remark this answers is not in the document]
Comment (Cabraham1) [reply src/modules/likesCounter.js:37]: done

File: src/modules/itemCounter.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+const menuCounter = async () => {
+  await fetch(API_URL)
+    .then((response) => response.json())
+    .then((data) => {
+      const meal = data.meals;
+      const count = meal.length;
+      itemCounter.innerHTML = `(${count})`;
+    });
+};
```
[the remark this answers is not in the document]
Comment (Cabraham1) [reply src/modules/itemCounter.js:12]: done

File: README.md (hunk 6 above)
Comment (HAMMAS-SALEEM) [reply README.md]: ## 👍

File: src/index.js [hunk 9]
```diff
@@ -1 +1,14 @@
...
+import menuCounter from './modules/itemCounter.js';
+import { likes, displayLikes } from './modules/likesCounter.js';
+import './assets/close.svg';
+
+display();
+likes();
+displayLikes();
+menuCounter();
+const niceMeal = new Meal();
+niceMeal.getMeal();
```
Comment (HAMMAS-SALEEM) [inline src/index.js:14]: ## Before
image

## After
image

## [OPTIONAL]
- [ ] Kudos on implementing the async await 👏 but I guess while updating the `displayLikes()` method you broke something in your application 😨 When I reload your page your application won't get likes but when I like any recipe then your app gets like for only that recipe 🤔 Kindly fix it 😊

You can fix it by calling the `displayLIkes()` method at the end of your index.js file 😊 like below

image


Now you are thinking I should call it once and I should remove the callback at line 11 but if you do it then it won't work. You have to keep your code same and call the displayLikes() method at the end 😊

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

### Hey @Cabraham1, @Jord4N-AC 👋

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow (For Development)
✔️ Following GitHub Flow (For External Review)
✔️ Following Requirements

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: MIT.md [hunk 10]
```diff
@@ -1,10 +1,10 @@
-## Copyright 2021, [YOUR NAME]
```
Comment (HAMMAS-SALEEM) [inline MIT.md:1]: image

**[OPTIONAL]**

- [ ] You don't have a descriptive pull request summary 😨 Kindly add a descriptive pull request summary to make your pull request descriptive 😊

**Comment by Cabraham1** [comment]
@HAMMAS-SALEEM 

I wanted to take a moment to express my sincere gratitude for the time and effort you put into reviewing and approving my code. Your insights and suggestions have been incredibly helpful in improving my work and ensuring that it meets the required standards.

Your feedback has not only helped me to improve the quality of my code but has also expanded my knowledge of the programming language and best practices. The optional suggestions you provided were especially valuable, and I appreciate the extra effort you put into providing them.

Your approval of my code has boosted my confidence and has given me a great sense of accomplishment. I feel fortunate to have had the opportunity to work with someone as knowledgeable and experienced as you, and I hope to continue learning from you in the future.

Thank you once again for all of your help, support, and guidance. I look forward to working with you again soon.
@Cabraham1

File: src/index.js (hunk 9 above)
Comment (Cabraham1) [reply src/index.js:14]: thanks

File: MIT.md (hunk 10 above)
Comment (Cabraham1) [reply MIT.md:1]: right away 😉

**Comment by HAMMAS-SALEEM** [comment]
Hey, @Cabraham1 I am happy to be helpful.
Do let me know if you have any questions or confusion. You can reach out to me on slack by searching my name **Hammas** in **full-time-students** channel :blush:

**Review by rotimiazeez — COMMENTED** [review]
Hi @Cabraham1 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good App UI 🖼️
- Good Test implementation ✔️
- Descriptive README ✔️
- Excellent communication with the CRs ✔️
- Professional commit messages ✔️
- No linter error ✔️

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend considering them as they can improve your code. Some of them were missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something needs to be 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**